gcd/src/main/java/org/spine3/server/storage/datastore/DsRecordStorage.java

```diff
@@ -259,21 +267,33 @@ protected EntityQuery buildAllQuery(TypeUrl typeUrl) {
         return query;
     }
 
-    protected Entity entityRecordToEntity(I id, EntityRecord record) {
+    protected Entity entityRecordToEntity(I id, EntityRecordWithColumns record) {
+        final EntityRecord entityRecord = record.getRecord();
         final Key key = keyFor(datastore,
-                               kindFrom(record),
+                               kindFrom(entityRecord),
                                ofEntityId(id));
-        final Entity incompleteEntity = Entities.messageToEntity(record, key);
-
+        final Entity incompleteEntity = Entities.messageToEntity(entityRecord, key);
         final Entity.Builder entity = Entity.newBuilder(incompleteEntity);
-        entity.set(VERSION_KEY, record.getVersion()
-                                      .getNumber());
+
+        populateFromStorageFields(entity, record);
+
         final Entity completeEntity = entity.build();
         return completeEntity;
     }
 
+    protected void populateFromStorageFields(Entity.Builder entity, EntityRecordWithColumns record) {
+        final ColumnTypeRegistry<? extends ColumnType<?, ?, Entity.Builder, String>> columnTypeRegistry =
```

Comment:
Wouldn't it be easier to declare the `columnTypeRegistry` class field as `ColumnTypeRegistry<? extends ColumnType<?, ?, Entity.Builder, String>>` and have this cast somewhere else? It is going to be much more of a fail-fast approach, than this.

Just a thought.